src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -1780,7 +1780,13 @@ def update_container_settings(cmd, resource_group_name, name, docker_registry_se
 
 def update_container_settings_functionapp(cmd, resource_group_name, name, registry_server=None,
                                           image=None, registry_username=None,
-                                          registry_password=None, slot=None, min_replicas=None, max_replicas=None):
+                                          registry_password=None, slot=None, min_replicas=None, max_replicas=None,
+                                          workload_profile_name=None, cpu=None, memory=None):
+    if is_centauri_functionapp(cmd, resource_group_name, name):
+        _validate_cpu_momory_functionapp(cpu, memory)
+        update_resource_config(cmd, resource_group_name, name, cpu=cpu, memory=memory,
+                               workload_profile_name=workload_profile_name)
```

Comment:
Can you add a check here to make sure that the updates are only made if `workload_profile_name`, `cpu` or `memory` are updated?